### PR TITLE
fix home-screen half-width

### LINKED ISSUE #14894: Home and Profile screens is sometimes displayed at half height
# Bug Report

## Problem 
The only known thing atm is that both times the issue was reproduced, there were at least three chats on the `Chat` screen: one group and two 1-1 chats, and one community on the home screen (this may have been a coincidence though). Also, the bug was reproduced after login / return from the background.

The video does not show that the `Profile` screen also began to be displayed at half height.

If you create a new multi-account, the issue persists.

Screenshot 2023-01-25 at 16 56 10

https://user-images.githubusercontent.com/67952253/214611542-7c4a135a-f534-4ad1-871e-d1abf3b74523.mp4

### Reproduction

images

### Additional Information

- Status version: nightly 25/01
- Operating System: Android

[comment]: # (Please, add logs/notes if necessary)

### PR DESCRIPTION
**Pull request by Parveshdhull**
fixes https://github.com/status-im/status-mobile/issues/14894

### Summary

For fixing https://github.com/status-im/status-mobile/issues/14633, in https://github.com/status-im/status-mobile/pull/14639 we used the height provided by the rendered layout as screen height.
I guess for some reason in a very rare case height calculated by on-layout was half of the total height.
(probably due to screen rotation) (even though we only have portrait orientation🤔)

1. So PR makes sure that height is more than width before storing
2. Also removed the `(when-not @animation/screen-height` condition, because it is small overhead, but better than the user getting stuck at one stored value(even if value is wrong)
3. Make sure height returned by on-layout is more than height returned by dimensions

### Testing
Please test https://github.com/status-im/status-mobile/issues/14633, screen height for Xiaomi devices

Thank you @qoqobolo for finding this

status: ready

### CODE REVIEW

File: src/status_im2/contexts/shell/view.cljs [hunk 1]
```diff
@@ -117,13 +117,26 @@
...
-     (let [shared-values (animation/calculate-shared-values)]
+     (let [shared-values (animation/calculate-shared-values)
+           dimensions    (rf/sub [:dimensions/window])]
        (animation/change-root-status-bar-style)
        [rn/view
         {:style     {:flex 1}
-         :on-layout (when-not @animation/screen-height
-                      (fn [evt]
-                        (let [height (oget evt "nativeEvent" "layout" "height")]
+         :on-layout (fn [evt]
```
Comment (J-Son89) [inline src/status_im2/contexts/shell/view.cljs]: consider moving to a helper function 👍

**Review by J-Son89 — APPROVED** [review]

File: src/status_im2/contexts/shell/view.cljs [hunk 2]
```diff
@@ -117,13 +117,26 @@
...
+                        ;; screen is horizontal
+                        ;; 2. Initialize values with 0 in case of nil
+                        ;; 3. In the case of notch devices, the dimensions height will be smaller than
+                        ;; on-layout,
+                        ;; (without status bar height included)
+                        ;; https://github.com/status-im/status-mobile/issues/14633
+                        ;; 4. In the case of devices without a notch, both heights should be the same,
+                        ;; but actual values differ in some pixels, so arbitrary 5 pixels is allowed
+                        (when (and (> height width)
+                                   (>= (+ height 5) (or (:height dimensions) 0)))
```
Comment (flexsurfer) [inline src/status_im2/contexts/shell/view.cljs]: +5 ?

Comment (Parveshdhull) [reply src/status_im2/contexts/shell/view.cljs]:
```
;; 4. In the case of devices without a notch, both heights should be the same,
;; but actual values differ in some pixels, so arbitrary 5 pixels is allowed
```
better safe than sorry :)

File: src/status_im2/contexts/shell/view.cljs (hunk 1 above)
Comment (Parveshdhull) [reply src/status_im2/contexts/shell/view.cljs]: done, thank you

**Comment by qoqobolo** [comment]
Hi @Parveshdhull !
Something's off with iOS builds, could you check please?

**Comment by churik** [comment]
@Parveshdhull tested on Xiaomi mi 9T pro, works the same way as in nightly, so height re-calculates after relogin, so LGTM.
Thank you!

**Comment by qoqobolo** [comment]
Works fine on my devices too.
Let's merge, I'll let you know if the issue reappears at some point. 
@Parveshdhull thank you for finding a possible cause almost blindly and fixing it!

**Pull request merged**